src/HtmlProcessor.cs

```diff
@@ -70,11 +147,20 @@ public static void ProcessFile(string inputPath, string outputPath)
             try
             {
                 string text = File.ReadAllText(inputPath);
-                string html = ConvertTextToHtml(Path.GetFileNameWithoutExtension(inputPath), text);
-
-                string outputFileName = Path.Combine(outputPath, Path.GetFileNameWithoutExtension(inputPath) + ".html");
-                File.WriteAllText(outputFileName, html);
-
+                string html = "\0";
+
+                if (Path.GetExtension(inputPath) == ".txt")
+                {
+                    html = ConvertTextToHtml(Path.GetFileNameWithoutExtension(inputPath), text);
+
+                }
+                if (Path.GetExtension(inputPath) == ".md")
+                {
+                    html = ConvertMdToHtml(Path.GetFileNameWithoutExtension(inputPath), text);
+                }
```

Comment:
store the extension in a variable on this line, and use the variable for the following if checks. That way, we don't have to call the GetExtension function twice.
Alternatively, change the block to an if-else statement, because a file can only be .txt or .md. If it's not one, it's the other. Since by the time we have reached this point in the code, we have already checked for those two. No other file type would make the code reach this line.

Reply:
Went with the if-else statement instead.